src/dotnet/ToolPackage/LockFileMatcher.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class LockFileMatcher
+    {
+        private static string[] _pathInLockFilePathInArray;
+        private static string[] _entryPointPathInArray;
+
+        internal static bool MatchesFile(string pathInLockFile, string entryPoint)
+        {
+            _pathInLockFilePathInArray = SplitPathByDirectorySeparator(pathInLockFile);
```

Comment:
Why the use of static fields here instead of locals?

Reply:
Please see if the local function + strong type input + comments is good enough